static/js/util/api.js

```diff
@@ -62,6 +66,10 @@ function fetchJSONWithCSRF(input, init) {
     // Also note that text is a promise here, not a string
     let text = response.text();
 
+    // For 400 and 401 errors, force login
```

Comment:
the 400 is also an error we get from edX in a special case: when we send a refresh for a token without a refresh token string. This is a very edge case, I admit it.